components/ias-adapter/internal/service/ias/service.go

```diff
@@ -0,0 +1,181 @@
+			errors.Newf("failed to update IAS application '%s' with UCL ID '%s': %w", data.ConsumerApplication.ID, data.TenantMapping.AssignedTenants[0].UCLApplicationID, err)
+		}
+	}
+
+	return nil
+}
+
+func (s Service) GetApplication(ctx context.Context, iasHost, clientID string) (types.Application, error) {
+	log := logger.FromContext(ctx)
+
+	timeoutCtx, cancel := context.WithTimeout(ctx, s.cfg.RequestTimeout)
+	defer cancel()
+
+	url := buildGetApplicationURL(iasHost, clientID)
+	req, err := http.NewRequestWithContext(timeoutCtx, http.MethodGet, url, nil)
+	if err != nil {
+		return types.Application{}, errors.Newf("failed to create request: %w", err)
+	}
+	resp, err := s.client.Do(req)
+	if err != nil {
+		return types.Application{}, errors.Newf("failed to send request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		respBytes, err := io.ReadAll(resp.Body)
+		if err != nil {
+			log.Warn().Msgf("failed to read GET application response body: %s", err)
+		}
+		return types.Application{}, errors.Newf("failed to get application, status '%d', body '%s'", resp.StatusCode, respBytes)
+	}
+
+	applications := types.Applications{}
+	if err := json.NewDecoder(resp.Body).Decode(&applications); err != nil {
+		return types.Application{}, err
+	}
+
+	if len(applications.Applications) == 0 {
+		return types.Application{}, errors.Newf("application with clientID '%s' not found", clientID)
+	}
+	if len(applications.Applications) > 1 {
+		return types.Application{}, errors.Newf("found more than one application with clientID '%s'", clientID)
+	}
+
+	return applications.Applications[0], nil
+}
+
+func addConsumedAPI(consumedAPIs []types.ApplicationConsumedAPI, consumedAPI types.ApplicationConsumedAPI) []types.ApplicationConsumedAPI {
+	for _, api := range consumedAPIs {
+		if api.APIName == consumedAPI.APIName {
+			return consumedAPIs
+		}
+	}
+	return append(consumedAPIs, consumedAPI)
+}
+
+func removeConsumedAPI(consumedAPIs []types.ApplicationConsumedAPI, apiName string) []types.ApplicationConsumedAPI {
+	found := false
+	i := -1
+	for i = range consumedAPIs {
+		if consumedAPIs[i].APIName == apiName {
+			found = true
+			break
+		}
+	}
+	if !found {
+		return consumedAPIs
+	}
+	consumedAPIs[i] = consumedAPIs[len(consumedAPIs)-1]
+	return consumedAPIs[:len(consumedAPIs)-1]
+}
+
+func (s Service) updateApplication(ctx context.Context, iasHost, applicationID string,
+	consumedAPIs []types.ApplicationConsumedAPI) error {
+
+	log := logger.FromContext(ctx)
+
+	appUpdate := types.ApplicationUpdate{
+		Operations: []types.ApplicationUpdateOperation{
+			{
+				Operation: types.ReplaceOp,
+				Path:      types.ConsumedAPIsPath,
+				Value:     consumedAPIs,
+			},
+		},
+	}
+	appUpdateBytes, err := json.Marshal(appUpdate)
+	if err != nil {
+		return errors.Newf("failed to marshal body: %w", err)
+	}
+	url := buildPatchApplicationURL(iasHost, applicationID)
+	timeoutCtx, cancel := context.WithTimeout(ctx, s.cfg.RequestTimeout)
```

Comment:
Would be nice to encapsulate the timeout in a method - in this way will be written once and we will not forget adding it. E.g., have a dedicated _Client_ struct with and `http.Client` as a field which handles the timeout.